packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -189,7 +188,7 @@ class Xcode {
 Future<XcodeBuildResult> buildXcodeProject({
   BuildableIOSApp app,
   BuildInfo buildInfo,
-  String target: bundle.defaultMainPath,
+  String target,
```

Comment:
This is good. Though I think reading the flow downstream, there might be some issues since xcodeproject.dart's updateGeneratedXcodeProperties always writes a line FLUTTER_TARGET which will be 'null'. In xcode_backend.sh, the variable $FLUTTER_TARGET will exist and turn off the default value of 'lib/main.dart' I think. 

Maybe we can make these names more explicit to avoid issues for future maintainers. 

Let's rename `target` to `targetOverride` here in the `buildXcodeProject`'s signature and in `updateGeneratedXcodeProperties`. Let `updateGeneratedXcodeProperties` not write anything into the generated file if `targetOverride` is null. 

Let's add a dartdoc for this method to say that targetOverride is optional and if it's null, the value default file taken by xcode_backend.sh is lib/main.dart.

Reply:
@xster My reason for removing this was that it was cause of the bug in the first place, the default parameter masked the need for it to be specified. I've applied the suggested changes.

But this leads to another place in the flutter tool where it is used as a default parameter. That might also want to be cleared up. https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/bundle.dart#L32

Even though we have other sane defaults there. It is a case that the upstream always supplies the default value down for the mainPath parameter and nowhere in the code is it not currently specified (once this PR get's merged).